teps/0091-trusted-resources.md

```diff
@@ -141,25 +137,37 @@ In later iterations, this can be extended such that tag-referenced images can
 _themselves_ be verified for the overall Task to pass verification. However,
 this will be left out of the scope of this TEP.
 
-The Pipeline Author will be able to provide optional configuration to Tekton
-Chains such that Chains will *only* consider results from Tasks that were
-verifiable (and, following the previous rule, were successfully verified).
-
 
 ### Risks and Mitigations
 
-* Mutating webhook is a risk of the proposal. The proposed verification will happen after the mutating webhook, so the content of a Resource may be mutated and fail the verification. This would be an issue for local cluster resources.
+* Mutating webhook is a risk of the proposal. The proposed verification will happen after the mutating webhooks, so the content of a Resource may be mutated and fail the verification. This would be an issue for local cluster resources. And it also means that if we don't address this issue, all signed resources need to be signed again everytime we change the mutating webhook.
 
-  Some possible mitigations include the following:
+  Some possible mitigations/solutions include the following:
 
   1. Verify the Tasks/Pipelines when applying them in the cluster, and update the Resources to add the missing mutating values (e.g. Use `SetDefaults`) everytime we bump the version of Tekton Pipeline.
   2. Avoid mutating Tekton Resources or move the mutating after verification.
   3. Store the encoded resource in annotation before mutating and verify the stored resource at validating webhook.
   4. Do verification in mutating webhook, if failed verification then mark the run as failed and later check and reject the resource in validationg webhook.
 
+**Pros vs Cons:**
+| Option | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Call setdefaults when signing | Help users add missing values |  Need to update signature when there is update in mutating
```

Comment:
An issue I can see with this is it effectively ties client versions to server versions. This could cause conflicts with API compatibility since any change to SetDefaults that would cause a change in the signature would now be a backwards incompatible change.